dom-manipulation/main.js

```diff
@@ -0,0 +1,23 @@
+function buttonClickCounter() {
+  let clickCount = 0;
+  const hotBtn = document.querySelector('.hot-button');
+  const clickCounter = document.querySelector('.click-count');
+  hotBtn.addEventListener('click', function () {
+    clickCount++;
+    clickCounter.textContent = 'Clicks: ' + clickCount;
+    if (clickCount <= 4) {
+      hotBtn.className = 'hot-button cold';
+    } else if (clickCount > 4 && clickCount <= 7) {
```

Comment:
No need for the logical && operator in this case because these else if blocks will only run after the previous condition if it is not true.